packages/edit-navigation/src/utils/index.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * The purpose of this function is to create a middleware that is responsible for preloading menu-related data.
+ * It uses data that is returned from the /__experimental/menus endpoint for requests
+ * to the /__experimental/menu/<menuId> endpoint, because the data is the same.
+ * This way, we can avoid making additional REST API requests.
+ * This middleware can be removed if/when we implement caching at the wordpress/data level.
+ *
+ * @param {Object} preloadedData
+ * @return {Function} Preloading middleware.
+ */
+export function createMenuPreloadingMiddleware( preloadedData ) {
+	const cache = Object.keys( preloadedData ).reduce( ( result, path ) => {
+		result[ getStablePath( path ) ] = preloadedData[ path ];
+		return result;
+	}, /** @type {Record<string, any>} */ ( {} ) );
+
+	let menusDataLoaded = false;
+	let menuDataLoaded = false;
+
+	return ( options, next ) => {
+		const { parse = true } = options;
+		if ( 'string' !== typeof options.path ) {
+			return next( options );
+		}
+
+		const method = options.method || 'GET';
+		if ( 'GET' !== method ) {
+			return next( options );
+		}
+
+		const path = getStablePath( options.path );
+		if ( ! menusDataLoaded && cache[ path ] ) {
+			menusDataLoaded = true;
+			return sendSuccessResponse( cache[ path ], parse );
+		}
+
+		if ( menuDataLoaded ) {
+			return next( options );
+		}
+
+		const matches = path.match(
+			/^\/__experimental\/menus\/(\d+)\?context=edit$/
+		);
+		if ( ! matches ) {
+			return next( options );
+		}
+
+		const key = Object.keys( cache )?.[ 0 ];
+		if ( ! key ) {
+			return next( options );
+		}
+
+		const menuData = cache[ key ]?.body;
+		if ( ! menuData ) {
+			return next( options );
+		}
+
+		const menuId = parseInt( matches[ 1 ] );
+		const menu = menuData.filter( ( { id } ) => {
+			return id === menuId;
+		} );
+
+		if ( 0 < menu.length ) {
```

Comment:
Yoda conditions don't read well, e.g. here it's "if zero is less than menu length then." I know there are some guidelines to do that on PHP side to prevent accidental assignments like `$a = 4` instead of `$a === 5`, but these days linters are pretty good at catching that (and also it doesn't apply to `<`).

```suggestion
		if ( menu.length ) {
```

Reply:
> Yoda conditions don't read well,

I see what you mean. I can agree it can be harder to read Yoda conditions for some people.
Although Yoda conditions have their advantages and it's not a crime to use them. IMO it's just a personal preference.

`if ( value > 0 )` is better than just `if ( value )`, because the former is a stricter comparison.
If `value ===  -1`, the latter expression will return `true`, which is not what it is supposed to do.
This can happen if we compare some unsigned numbers like bank account amounts.
I understand that length of the array cannot be lower than zero. But I prefer to write it like that everywhere because it will work correctly regardless of whether we compare unsigned or signed integers.

Reply:
Fixed (?) in 612d17b6d1a07a83db7bf35173efddc14e1b9897